runtime/lib/mirrors_impl.dart

```diff
@@ -1676,4 +1681,18 @@ class _Mirrors {
     }
     return typeMirror;
   }
+
+  static Type _resolveType(Type key, List<Type> typeArguments) {
+    if (!typeArguments.every((_) => _ is Type)) {
```

Comment:
Figured out how to do this in C: 5fdc5b9 ( [actual line(s)](https://github.com/dart-lang/sdk/pull/26012/files#diff-627364e40d12c87611f09ab984c5c521R872) ).
Sanity check from @rmacnak-google would be great though!